homeassistant/components/google_air_quality/sensor.py

```diff
@@ -0,0 +1,186 @@
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_BILLION,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import GoogleAirQualityConfigEntry
+from .const import DOMAIN
+from .coordinator import GoogleAirQualityUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class AirQualitySensorEntityDescription(SensorEntityDescription):
+    """Describes Air Quality sensor entity."""
+
+    exists_fn: Callable[[AirQualityData], bool] = lambda _: True
+    value_fn: Callable[[AirQualityData], StateType | datetime]
+
+
+AIR_QUALITY_SENSOR_TYPES: tuple[AirQualitySensorEntityDescription, ...] = (
+    AirQualitySensorEntityDescription(
+        key="uaqi",
+        translation_key="uaqi",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.AQI,
+        value_fn=lambda x: x.indexes[0].aqi,
+    ),
+    AirQualitySensorEntityDescription(
+        key="uaqi_category",
+        translation_key="uaqi_category",
+        device_class=SensorDeviceClass.ENUM,
+        options=[
+            "excellent_air_quality",
+            "good_air_quality",
+            "low_air_quality",
+            "moderate_air_quality",
+            "poor_air_quality",
+        ],
+        value_fn=lambda x: x.indexes[0].category,
+    ),
+    AirQualitySensorEntityDescription(
+        key="local_aqi",
+        translation_key="local_aqi",
+        exists_fn=lambda x: x.indexes[1].aqi is not None,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.AQI,
+        value_fn=lambda x: x.indexes[1].aqi,
+    ),
+    AirQualitySensorEntityDescription(
+        key="local_category",
+        translation_key="local_category",
+        device_class=SensorDeviceClass.ENUM,
+        value_fn=lambda x: x.indexes[1].category,
+    ),
+    AirQualitySensorEntityDescription(
+        key="uaqi_dominant_pollutant",
+        translation_key="uaqi_dominant_pollutant",
+        device_class=SensorDeviceClass.ENUM,
+        value_fn=lambda x: x.indexes[0].dominant_pollutant,
+    ),
+    AirQualitySensorEntityDescription(
+        key="local_dominant_pollutant",
+        translation_key="local_dominant_pollutant",
+        device_class=SensorDeviceClass.ENUM,
+        value_fn=lambda x: x.indexes[1].dominant_pollutant,
+    ),
```

Comment:
enum is worthless wtihout options (note, this is not the cue to remove the enum device class, but to add options)

Reply:
As this integration implements an universal AQI at list position `0` and and local AQI of the selected country (here at list position `1`), each of the round about hundred countries has different options.
1) So we can either type all the options for each country by hand in the library, or in home assistant 
2) drop the enum. 

Option 2 has also the benefit, that all the translation strings in `strings.json` could be removed. And we could get localised strings directly from the API

Reply:
I would suggest option 2. It reduces a lot of work and possibility for errors.

Reply:
https://github.com/home-assistant/core/pull/145237/commits/da5e339aa703cd06d75e4bfb315128cc2db950d7